common/src/stack/command/stack/commands/create/mirror/__init__.py

```diff
@@ -150,15 +150,23 @@ def mirror(self, mirror_url, quiet):
 		os.symlink(url, 'RPMS')
 
 	def repoquery(self, repoid, repoconfig):
-		cmd = 'repoquery -qa --repoid=%s' % repoid
-
-		if repoconfig:
-			cmd = cmd + ' --config=%s' % repoconfig
-
-		proc = subprocess.Popen(shlex.split(cmd), stdin=None,
-			stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-		o, e = proc.communicate()
-		return o, e
+                #Check if running on sles and through error if using repoid
+                if self.os == "sles" and repoid != None :
+                        print( "sles does not have repoquery package available.\nPlease use URL for creating pallets")
+                        return "None", "None"
```

Comment:
I think this should just raise a `CommandError` with the printed string as the `msg`. That way, execution of the command stops, and the normal error printing mechanisms can take over. This also seems like a good error case which warrants an exception.

Reply:
OK thanks. I thought I had updated the develop branch on my fork but I guess i didn't rebase the bugfix branch on it. I'll have to look into it again pretty new to actually using git. I will do some python learning to implement the error as described

Reply:
It would probably be sufficient to do something like:
```
if self.os == "sles" and repoid != None :
    raise CommandError(
        cmd = self,
        msg = "sles does not have repoquery package available.\nPlease use URL for creating pallets",
    )
```